### PR TITLE
Balance clean/refactoring part 1

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/wallet/account/operations/balance.rs [hunk 1]
```diff
@@ -30,108 +34,98 @@ impl AccountHandle {
...
-        for output_data in account.unspent_outputs.values() {
+        let relevant_unspent_outputs = account
+            .unspent_outputs
+            .values()
             // Check if output is from the network we're currently connected to
-            if output_data.network_id != network_id {
-                continue;
-            }
+            .filter(|data| data.network_id == network_id)
+            .map(|data| (&data.output_id, &data.output));
```
Comment (Thoralf-M) [inline sdk/src/wallet/account/operations/balance.rs:47]: filter_map()?

Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:47]: Sure, weird that clippy didn't find this

Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:47]: Thoralf > Clippy

Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:47]: Hmm actually not sure, you need something that returns an option for filter_map

Comment (Thoralf-M) [reply sdk/src/wallet/account/operations/balance.rs:47]: That's why the question mark, leave it like it is for now

**Review by Thoralf-M — DISMISSED** [review]
Changelog, fields aren't public anymore :trollface: Can be done in the next PR

File: sdk/src/wallet/account/operations/balance.rs [hunk 2]
```diff
@@ -19,6 +17,12 @@ impl AccountHandle {
     /// Get the AccountBalance
     pub async fn balance(&self) -> crate::wallet::Result<AccountBalance> {
         log::debug!("[BALANCE] get balance");
+        let mut account_balance = AccountBalance::default();
+        #[cfg(feature = "participation")]
+        {
+            account_balance.base_coin.voting_power = self.get_voting_power().await?;
```
Comment (DaughterOfMars) [inline sdk/src/wallet/account/operations/balance.rs:23]: Does this need a changelog entry?

File: sdk/src/wallet/account/operations/balance.rs [hunk 3]
```diff
@@ -248,15 +239,13 @@ impl AccountHandle {
 
         log::debug!(
             "[BALANCE] total_amount: {}, locked balance: {}, total_rent_amount: {}",
```
Comment (Thoralf-M) [inline sdk/src/wallet/account/operations/balance.rs]:
```suggestion
            "[BALANCE] total_amount: {}, locked_amount: {}, total_rent_amount: {}",
```

File: sdk/src/wallet/account/operations/balance.rs [hunk 4]
```diff
@@ -30,108 +34,98 @@ impl AccountHandle {
...
         let mut total_native_tokens = NativeTokensBuilder::new();
-        let mut potentially_locked_outputs = HashMap::new();
-        let mut aliases = Vec::new();
-        let mut foundries = Vec::new();
-        let mut nfts = Vec::new();
 
         let account = self.read().await;
 
-        for output_data in account.unspent_outputs.values() {
+        let relevant_unspent_outputs = account
```
Comment (DaughterOfMars) [inline sdk/src/wallet/account/operations/balance.rs:42]: Could just move this below into the loop?

File: sdk/src/wallet/account/operations/balance.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:23]: Not really, was already done here https://github.com/iotaledger/iota-sdk/pull/243/files#diff-cac9231564f0d596c25a156c4d2020ae6aba9c03e27eb430e78c0d2c32ecec2fL294, just been re-arranged

File: sdk/src/wallet/account/operations/balance.rs [hunk 5]
```diff
@@ -30,108 +34,98 @@ impl AccountHandle {
...
+            .unspent_outputs
+            .values()
             // Check if output is from the network we're currently connected to
-            if output_data.network_id != network_id {
-                continue;
-            }
+            .filter(|data| data.network_id == network_id)
```
Comment (DaughterOfMars) [inline sdk/src/wallet/account/operations/balance.rs:46]: `filter_map` maybe?

Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:46]: It's not that straightforward because you'd need an option inside, what would you do ?

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/wallet/account/operations/balance.rs (hunk 5 above)
Comment (DaughterOfMars) [reply sdk/src/wallet/account/operations/balance.rs:46]: I tried it and, you're right, not worth it.

**Review by DaughterOfMars — APPROVED** [review]

File: sdk/src/wallet/account/operations/balance.rs (hunk 4 above)
Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/balance.rs:42]: You mean right after the `in` ? I had that at the beginning but I preferred a variable to be able to name it and make it a bit more obvious what it was

**Pull request merged**